### ISSUE TITLE
Prisma not working with sst monorepo

### ISSUE THREAD

**Issue by SchmellaKaa** [prisma/prisma#18570]
### Bug description

SST is a framework to build serverless applications on aws and with the update to version 2.0 switched to a monorepo setup.  Pre 2.0 it worked fine with prisma but since version 2.0 i ran into similar problems than other users have with nextjs.

They have an example setup with prisma where they have already a workaround applied to copy the prisma engines to the correct locations. This worked before v2.0 but now additional problems arise due to sst being a monorepo.

I already posted an [issue](https://github.com/serverless-stack/sst/issues/2667) on the sst git, but, as it is essentially a problem of prisma working with monorepos i decided to post it here aswell.

### How to reproduce

- Create an sst project using: `yarn create sst --template=examples/prisma`
- run `yarn install`
- run `yarn run dev`  // deploys lambdas structure to aws

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by dbibbos** [comment]
I had the same problem. Based on the following example, I performed some Prisma-related changes in our project, which worked out.

[SST Prisma example](https://github.com/geauser/sst-prisma-example/tree/master)

**Comment by jamiechong** [comment]
I'm getting similar issues but I'm on prisma 5.0.0. @dbibbos  what was the "prisma-related changes" you made? In the linked example, I don't really see much difference between your code and mine.

**Comment by perfectbase** [comment]
I have the same problem.. I'm also in the sst issue and in the discord trying to figure this out, but no luck yet.
I did see the example @dbibbos mentioned, but I couldn't find a way to use it for a monorepo setup (with turborepo).

here is an example app with my current setup:
https://github.com/perfectbase/prisma-examples

ps: In this example I'm trying with lambda layers, because that is what the official example was doing... but I'm ok with removing the lambda layer if that makes things simpler.

**Comment by mng-prophero** [comment]
Any update on this as we're upgrading a project from sst v1 to v2 and are now running into this issue as well...

**Comment by a-eid** [comment]
@jkomyno @janpio any updates on this ?

**Comment by janpio** [comment]
No, this is an unconfirmed bug report. If you have a reproduction of the same problem, please share it with us.
https://github.com/jetbridge/sst-prisma uses SST v2 afaik, so it seems to be possible sometimes / to some.

**Comment by a-eid** [comment]
yes we do, we have not been able to use prisma for sst for a while now. I'll try to make a minimal reproducible example for you today. would that help ?

**Comment by janpio** [comment]
Yes, that is exactly what is needed for us to be able to understand what might be going on.

**Comment by a-eid** [comment]
@janpio please check this repo https://github.com/a-eid/sst-prisma-issue

**Comment by janpio** [comment]
Which commands exactly?
Is there a way to test sst (and reproduce the problem) without having to provide AWS credentials?

**Comment by a-eid** [comment]
@janpio while playing around with this I get the `__filename is not defined` error referred on this repo currently.

**Comment by janpio** [comment]
Can you explain how exactly you "play around" to get this? I only get the one above.

**Comment by a-eid** [comment]
I mean while trying to get prisma to work with sst, I did get the above error ( that's how I landed on this issue ). I can not reproduce it right now. but if you would like I could try.

**Comment by millsp** [comment]
`__filename` does not exist in ESM hence the following error at runtime

> ReferenceError: __filename is not defined

Also confirmed by the fact that you are running in an ESM runtime

> at async ESMLoader.import (node:internal/modules/esm/loader:530:24)",

This will be fixed by https://github.com/prisma/prisma/issues/5030

**Comment by millsp** [comment]
If you are not running into the `__filename` issue, and you have the schema not found issue, then it is likely that you need to include the schema near your bundle as well as the correct engine.

**Comment by a-eid** [comment]
@millsp I do, however the issue does not go away.

**Comment by janpio** [comment]
> but if you would like I could try.

Yes please do, although we know _why_ this is happening (as Pierre posted) we need a clear reproduction that we can try to fix this and confirm it is fully fixed and does not just uncover the next problem.

**Comment by a-eid** [comment]
@janpio I just did reproduce it on the repo, kindly check and let me know if I could help in any other way. 
thanks.

**Comment by janpio** [comment]
Just to confirm, you are still working based on your reproduction at https://github.com/a-eid/sst-prisma-issue?

**Comment by a-eid** [comment]
@janpio yes, it contains the code before the 'UPDATE', which reproduces the original issue mentioned `__filename is not defined`

**Comment by a-eid** [comment]
@janpio I've pushed a fix for `__filename is not defined` in development, however I'm getting an `Internal Server Error` after deployment. which I can't understand why it happens.

**Comment by janpio** [comment]
What does `in development` and `after deployment` mean in SST context for you?

**Comment by a-eid** [comment]
@janpio when I run `pnpm dev` ( live lambda development ), everything works. 
after I deploy `pnpm run deploy --stage staging`, I get an internal server error when calling `new PrismaClient()`. 

sadly I am unable to debug it however, am trying however.

**Comment by a-eid** [comment]
I assume more files need to be copied over, what do you think ?

**Comment by janpio** [comment]
I have no idea. Please properly describe the problem, link to the state of the code that has this problem, and give us some time to figure it out. Just trying random stuff will no lead to success here.

**Comment by a-eid** [comment]
@janpio I think I got it work ... finally.

**Comment by dylansproule** [comment]
Doesn't seem to matter which sample I try, always end up in this state.

Error: ENOENT: no such file or directory, lstat 'node_modules/.prisma'

This never gets generated, any thoughts?

**Comment by hugo082** [comment]
The issue comes from the engine binary resolution.
https://github.com/prisma/prisma/blob/7aac24cb0fdb7e10aea5cfc59df3fc951fd93212/packages/client/src/runtime/core/engines/common/resolveEnginePath.ts#L100

You can skip the resolution by providing the value statically via `PRISMA_QUERY_ENGINE_LIBRARY` / `PRISMA_QUERY_ENGINE_BINARY` environment variable.

As long term fix maybe the filename should be loaded via `eval('__filename')`? The solution seems to work for the dirname a few lines above
https://github.com/prisma/prisma/blob/7aac24cb0fdb7e10aea5cfc59df3fc951fd93212/packages/client/src/runtime/core/engines/common/resolveEnginePath.ts#L90

**Comment by ravibanks54** [comment]
I've also started running into this issue and it's been a major blocker for me. @hugo082 It looks like your solution might be the best - how did you determine the location of the binary?

**Comment by millsp** [comment]
Hey @ravibanks54 we would welcome a reproduction repo from your side so we can make sure we understood the issue correctly. Thanks!

**Comment by ravibanks54** [comment]
Hey, I ended up switching to Drizzle for now but i think ESM support would fix this issue

**Comment by sommeeeer** [comment]
@a-eid are you still able to get this to work? :) and what repo should i use?

**Comment by richard523** [comment]
I've got it working but has weird bugs with database, next auth, and jwt. I can login/logout on my localhost. When I push to staging environment, I end up only being able to login. Unable to log out. I'm unable to clearly understand where Prisma client lives when pushed to AWS using SST.
